modules/configuration/src/main/java/org/apache/ignite/configuration/ConfigurationChanger.java

```diff
@@ -210,19 +212,42 @@ public void initialize(ConfigurationType storageType) {
         }
     }
 
-    /** Temporary until the IGNITE-14372 */
-    public CompletableFuture<?> changeX(
-        List<String> path,
+    /** */
+    public CompletableFuture<Void> change(
         ConfigurationSource source,
         ConfigurationStorage storage
     ) {
-        assert path.isEmpty() : "Path support is not yet implemented.";
-
         SuperRoot superRoot = new SuperRoot(rootKeys);
 
         source.descend(superRoot);
 
-        return changeInternally(superRoot, storage);
+        Set<ConfigurationType> storagesTypes = new HashSet<>();
+
+        superRoot.traverseChildren(new ConfigurationVisitor<Object>() {
+            @Override public Object visitInnerNode(String key, InnerNode node) {
+                RootKey<?, ?> rootKey = rootKeys.get(key);
+
+                return storagesTypes.add(rootKey.type());
+            }
+        });
+
+        assert !storagesTypes.isEmpty();
+
+        if (storagesTypes.size() != 1) {
+            return CompletableFuture.failedFuture(
+                new ConfigurationChangeException("Cannot change configurations belonging to different storages.")
```

Comment:
How about message like `Cannot handle change request with configuration patches belonging to different storages.`?

Also it would be great if we can add some information about which configuration roots are involved in the situation (I think we can gather this information when traversing children of superRoot).

Reply:
I'll change the message. List of roots and more details will be added in future.